src/main/java/com/facebook/openwifirrm/modules/ProvMonitor.java

```diff
@@ -160,8 +194,79 @@ protected void syncDataToProv(
 			topo.values().stream().mapToInt(x -> x.size()).sum()
 		);
 
+		// Sync zone configs
+		deviceDataManager.updateZoneConfig(
+			configMap -> {
+				for (
+					Map.Entry<String, RRMDetails> entry : rrmDetails.entrySet()
+				) {
+					String venue = entry.getKey();
+					if (venue.isEmpty()) {
+						logger.error("Venue is blank for an RRM enabled zone");
+						continue;
+					}
+
+					String zone = venueIdToName.get(venue);
+					if (zone == null) {
+						logger.error(
+							"Venue name {} is not found in id mapping",
+							venue
+						);
+						continue;
+					}
+
+					DeviceConfig cfg = configMap
+						.computeIfAbsent(zone, k -> new DeviceConfig());
+					cfg.enableRRM =
+						cfg.enableConfig = cfg.enableWifiScan = true;
+
+					// read the details from the config
+					RRMDetails details = entry.getValue();
+					if (details == null) {
+						logger.error(
+							"No RRM details available for zone {} even though it has RRM enabled",
+							zone
+						);
+						continue;
+					}
+
+					cfg.schedule = new RRMSchedule();
+					if (details.rrm != null) {
+						cfg.schedule.cron = RRMScheduler
+							.parseIntoQuartzCron(details.rrm.schedule);
+						if (cfg.schedule.cron == null) {
+							logger.error(
+								"Invalid cron for zone {} ({}), not scheduling",
+								zone,
+								details.rrm.schedule
+							);
+							continue;
+						}
+
+						if (details.rrm.algorithms != null) {
+							cfg.schedule.algorithms =
+								new ArrayList<RRMAlgorithm>();
+							for (
+								RRMAlgorithmDetails algorithmDetails : details.rrm.algorithms
+							) {
+								cfg.schedule.algorithms.add(
+									RRMAlgorithm.parse(
+										algorithmDetails.name,
+										algorithmDetails.parameters
+									)
+								);
+							}
+						}
+						logger.info("Set RRM settings for zone {}", zone);
```

Comment:
I would create a sub-method in this class which builds `RRMSchedule` from `RRMDetails` input, or null upon failure.